File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        newSet = set(nums)

        ans = 0
        for num in nums:
            count = 1
            crawl = num - 1
            while crawl in newSet:
                count += 1
                newSet.remove(crawl)
                crawl -= 1
            crawl = num + 1
            while crawl in newSet:
                count += 1
                newSet.remove(crawl)
                crawl += 1
            ans = max(ans, count)
        return ans

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const set = new Set(nums);

    ans = 0;
    for (let val of set) {
        if (!set.has(val - 1)) {
            let cnt = 0;
            let num = val + 1;
            while (set.has(num)) {
                cnt++;
                num++;
            }
            ans = Math.max(ans, cnt + 1);
        }
    }

    return ans;
};